SETI/pds-opus
# 1257 allow selection of units in select metadata menu

- Fixes #1257 (implementation in select metadata menu)
- Were any Django, import pipeline, or table_schema files modified? N
  - If YES:
    - Database used: opus3_test_220906
    - All Django tests pass: Y
    - FLAKE8 run on modified code: Y
- Were any JavaScript or CSS files modified? Y
  - If YES:
    - JSHINT run on all affected files: Y
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: Y
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA
- Was the documentation reviewed for necessary changes or additions? Y
  - About
  - Getting Started
  - FAQ
  - API Guide
  - Tooltips

Description of changes:
- Add units with dropdown in selected metadata fields when:
    - The select metadata menu is open
    - A new field is added
        - Store current displayed unit and available units in the data attributes for each menu item so that later on when a new field is added, we can use these info to attached the unit dropdown to that specific field
- Style the dropdown
    - Set the max height
    - Enable scrolling
- Add the dropdown behaviors
    - Scroll on dropdown won't scroll the selected metadata field
    - Click dropdown item will:
        - Close the dropdown
        - Change the displayed unit 
        - Update the slug in opus.prefs.cols
            - Add ":" to specify unit, default unit will be used if it's not specified.
            - Check ":" when parsing slugs from opus.prefs.cols.  
        - Avoid page refreshing 
- Click "Save Changes" will update the metadata values based on selected units
